src/jit/instrxarch.cpp

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "jitpch.h"
+#include "instr.h"
+
+// clang-format off
+static const InstructionInfo instructionInfoArray[] = {
+#if DEBUG
+    #define INSTRUCTION(id, nm, fp, um, rf, wf, mr, mi, rm, a4, rr) \
+        {INS_##id, nm, {mr, mi, rm, a4, rr}, um, static_cast<InsFlag>((InsFlag_FloatingPoint*fp)|(InsFlag_ReadsFlags*rf)|(InsFlag_WritesFlags*wf))},
+#else
+    #define INSTRUCTION(id, nm, fp, um, rf, wf, mr, mi, rm, a4, rr) \
+        {{mr, mi, rm, a4, rr}, um, static_cast<InsFlag>((InsFlag_FloatingPoint*fp) | (InsFlag_ReadsFlags*rf) | (InsFlag_WritesFlags*wf))},
```

Comment:
So this single table is going to have a lot of BAD_CODE entries that didn't exist (and weren't needed) in the previous tables. E.g., all INST2 will waste 3 size_t entries. I'm not sure I'm worried about that, but I do generally worry about "de-optimization" unless there's a good reason for it.

Reply:
Today, there are:
* INST5: 9 definitions. 27/45 encodings are BAD_CODE
* INST4: 11 definitions. 3/44 encodings are BAD_CODE
* INST3: 390 definitions. 764/1170 encodings are BAD_CODE
* INST2: 24 definitions. 8/48 encodings are BAD_CODE
* INST1: 48 definitions. 0/48 encodings are BAD_CODE
* INST0: 35 definitions. 1/35 encodings are BAD_CODE

This change adds 1195 new BAD_CODE encodings with this change (size increase info is here: https://github.com/tannergooding)
* The cost can be made less on x64 by changing to size_t.

I believe that the overall increased readability here is better, especially with all the HWIntrinsic work that has required new instructions to be added (and with more instructions still to come)

It may be possible to reduce the cost further by separating the SIMD instructions (the vast bulk of the encodings, which are all `INST3` today, and which generally only use the `mr` and `rm` encodings) out from the general purpose instructions.

Reply:
> I believe that the overall increased readability here is better,

I don't see any increase in readability. More is usually not better. Or it's subjective.